index.js

```diff
@@ -467,6 +467,45 @@ Client.prototype.remove = function (path, version, callback) {
     );
 };
 
+/**
+ * Deletes a node and all its children.
+ *
+ * @param path {String} The node path.
+ * @param [version=-1] {Number} The version of the node.
+ * @param callback {Function} The callback function.
+ */
+Client.prototype.removeAll = function(path, version, callback) {
```

Comment:
I like the ZKUtil names better: `removeRecursively`, which makes it clear that it will remove the provided root node as well.